### PR TITLE
[full-ci] Move folder loading into service

### PR DESCRIPTION
**Pull request by dschmidt**
## Description
Move loading of folders from views to dedicated data providers.
For now this is a super-stupid 1:1 port, to reduce the danger to mess this up.

From my pov this is good enough for a first iteration. If you don't have major concerns with this approach, let's merge this asap and I will incorporate any feedback in https://github.com/owncloud/web/pull/6563. I would really like to avoid merge conflicts as this touches quite a few different places in the codebase :) 

## Related Issue

- Fixes <issue_link>

## Motivation and Context
1) Separation of ui and data logic is usually a good idea anyway
2) Make folder loading reusable (for apps e.g.), this will in the long run allow us to fix reload in the mediaviewer for non personal files contexts

## How Has This Been Tested?
This is a rather stupid refactor, if our test suite stays green, everything should be fine.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Futher cleanup

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oC10</strong> https://drone.owncloud.com/owncloud/web/23459/10/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23459/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-alice-2022-3-9-08-12-24.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23459/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-brian-2022-3-9-08-12-48.zip 


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oC10</strong> https://drone.owncloud.com/owncloud/web/23460/10/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23460/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-alice-2022-3-9-08-38-03.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23460/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-brian-2022-3-9-08-38-24.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23460/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-carol-2022-3-9-08-38-32.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23460/tracing/alice-shares-folder-with-file-to-brian-alice-2022-3-9-08-40-41.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23460/tracing/alice-shares-folder-with-file-to-brian-brian-2022-3-9-08-40-51.zip 


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oC10</strong> https://drone.owncloud.com/owncloud/web/23462/10/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23462/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-alice-2022-3-9-10-00-28.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23462/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-brian-2022-3-9-10-00-50.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23462/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-carol-2022-3-9-10-00-59.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23462/tracing/alice-shares-folder-with-file-to-brian-alice-2022-3-9-10-03-09.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23462/tracing/alice-shares-folder-with-file-to-brian-brian-2022-3-9-10-03-21.zip 


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oC10</strong> https://drone.owncloud.com/owncloud/web/23463/10/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23463/tracing/alice-shares-folder-with-file-to-brian-alice-2022-3-9-10-26-23.zip 


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oC10</strong> https://drone.owncloud.com/owncloud/web/23465/10/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23465/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-alice-2022-3-9-11-13-38.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23465/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-brian-2022-3-9-11-13-59.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/23465/tracing/alice-can-share-this-weeks-meal-plan-with-all-parents-carol-2022-3-9-11-14-08.zip 


</p></details>

**Comment by ownclouders** [comment]
Results for <strong>oCISTrashbinUploadMoveJourney</strong> https://drone.owncloud.com/owncloud/web/23466/69/1
The following scenarios passed on retry:
- webUITrashbinRestore/trashbinRestore.feature:115
- webUITrashbinRestore/trashbinRestore.feature:46

File: packages/web-app-files/src/services/folder/loaderProject.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+  public getTask(context: TaskContext): FolderLoaderTask {
+    return useTask(function* (signal1, signal2, ref, sameRoute, path = null) {
+      const router = context.router
+      const store = context.store
+
+      const graphClient = clientService.graphAuthenticated(
+        store.getters.configuration.server,
+        store.getters.getToken
+      )
```
Comment (dschmidt) [inline packages/web-app-files/src/services/folder/loaderProject.ts:23]: Can be moved out of the taskhagain

**Review by pascalwengerter — APPROVED** [review]
Fine with me as a transitional/temporary solution :) [sonar cloud finding](https://sonarcloud.io/project/issues?id=owncloud_web&pullRequest=6562&resolved=false&types=CODE_SMELL) seem legit, but also fine if they get resolved in the next iteration

**Pull request merged**